x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshotRepository.java

```diff
@@ -202,8 +202,7 @@ public Repository create(RepositoryMetaData metaData, Function<String, Repositor
                     throw new IllegalArgumentException(DELEGATE_TYPE.getKey() + " must be set");
                 }
                 Repository.Factory factory = typeLookup.apply(delegateType);
-                return new SourceOnlySnapshotRepository(factory.create(new RepositoryMetaData(metaData.name(),
-                    delegateType, metaData.settings()), typeLookup));
+                return new SourceOnlySnapshotRepository(factory.create(metaData, typeLookup));
```

Comment:
Let's mark this part as WIP, we can do a nicer thing so that in the e.g. cat repo list the type is correctly displayed for the source only repo (could just do this via an override to the `getMetaData` on the source only repo), I'd just like to get confirmation of the approach before doing that :)